pennylane/devices/qubit/preprocess.py

```diff
@@ -0,0 +1,220 @@
+    "Exp",
+    "Evolution",
+}
+
+
+def _stopping_condition(op):
+    """Specify whether or not an Operator object is supported by the device"""
+    return getattr(op, "has_matrix", False) and qml.matrix(op).shape[-1] % 2 == 0
+
+
+def _supports_observable(observable):
+    """Checks if an observable is supported by this device.
+
+    Args:
+        observable (type or str): observable to be checked
+
+    Returns:
+        bool: ``True`` iff supplied observable is supported
+    """
+    if isinstance(observable, type) and issubclass(observable, Observable):
+        observable = observable.__name__
+    if isinstance(observable, str):
+        return observable in _observables
+
+    return False
+
+
+def expand_fn(circuit, max_expansion=10):
+    """Method for expanding or decomposing an input circuit.
+
+    This method expands the tape if:
+
+    - nested tapes are present,
+    - any operations are not supported on the device.
+
+    Args:
+        circuit (.QuantumTape): the circuit to expand.
+        max_expansion (int): The number of times the circuit should be
+            expanded. Expansion occurs when an operation or measurement is not
+            supported, and results in a gate decomposition. If any operations
+            in the decomposition remain unsupported by the device, another
+            expansion occurs.
+
+    Returns:
+        .QuantumTape: The expanded/decomposed circuit, such that the device
+        will natively support all operations.
+    """
+
+    if any(isinstance(o, MidMeasureMP) for o in circuit.operations):
+        circuit = qml.defer_measurements(circuit)
+
+    if not all(_stopping_condition(op) for op in circuit.operations):
+        circuit = circuit.expand(depth=max_expansion, stop_at=_stopping_condition)
+
+    return circuit
+
+
+def batch_transform(circuit):
+    """Apply a differentiable batch transform for preprocessing a circuit
+    prior to execution.
+
+    By default, this method contains logic for generating multiple
+    circuits, one per term, of a circuit that terminates in ``expval(Sum)``.
+
+    .. warning::
+
+        This method will be tracked by autodifferentiation libraries,
+        such as Autograd, JAX, TensorFlow, and Torch. Please make sure
+        to use ``qml.math`` for autodiff-agnostic tensor processing
+        if required.
+
+    Args:
+        circuit (.QuantumTape): the circuit to preprocess
+
+    Returns:
+        tuple[Sequence[.QuantumTape], callable]: Returns a tuple containing
+        the sequence of circuits to be executed, and a post-processing function
+        to be applied to the list of evaluated circuit results.
+    """
+    expval_sum_in_obs = any(
+        isinstance(m.obs, Sum) and isinstance(m, ExpectationMP) for m in circuit.measurements
+    )
+
+    is_shadow = any(isinstance(m, ShadowExpvalMP) for m in circuit.measurements)
+
+    if expval_sum_in_obs and not is_shadow:
+        circuits, sum_fn = qml.transforms.sum_expand(circuit)
```

Comment:
We don't need to split the sum up.  We may need to do so with finite shots like with Hamiltonians, but we don't need it for now.